File: app/src/main/java/ru/geekbrains/notes/data/CardData.java
package ru.geekbrains.notes.data;

import android.os.Parcel;
import android.os.Parcelable;

import java.util.Date;

public class CardData implements Parcelable {

    private String id;
    private String title;
    private String text;

    private Date date;

    public CardData(String title, String text, Date date) {
        this.title = title;
        this.text = text;
        this.date = date;
    }

    protected CardData(Parcel in) {
        title = in.readString();
        text = in.readString();
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(title);
        dest.writeString(text);
    }

    @Override
    public int describeContents() {
        return 0;
    }

    public static final Creator<CardData> CREATOR = new Creator<CardData>() {
        @Override
        public CardData createFromParcel(Parcel in) {
            return new CardData(in);
        }

        @Override
        public CardData[] newArray(int size) {
            return new CardData[size];
        }
    };

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getText() {
        return text;
    }

    public void setText(String text) {
        this.text = text;
    }

    public Date getDate() {
        return date;
    }

    public String getId() {
        return id;
    }


    public void setDate(Date date) {
        this.date = date;
    }

    public void setId(String id) {
        this.id = id;
    }
}

File: app/src/main/java/ru/geekbrains/notes/data/CardData.kt
package ru.geekbrains.notes.data

import android.os.Parcel
import android.os.Parcelable
import android.os.Parcelable.Creator
import java.util.*

class CardData : Parcelable {
    var id: String? = null
    var title: String?
    var text: String?
    var date: Date? = null

    constructor(title: String?, text: String?, date: Date?) {
        this.title = title
        this.text = text
        this.date = date
    }

    protected constructor(`in`: Parcel) {
        title = `in`.readString()
        text = `in`.readString()
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(title)
        dest.writeString(text)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Creator<CardData> {
        override fun createFromParcel(parcel: Parcel): CardData {
            return CardData(parcel)
        }

        override fun newArray(size: Int): Array<CardData?> {
            return arrayOfNulls(size)
        }
    }
}